lib/sql/schema.sql

```diff
@@ -0,0 +1,121 @@
+    source_id INT DEFAULT NULL,
+    label_id INT DEFAULT NULL,
+    PRIMARY KEY(id)
+);
+
+CREATE TABLE
+    Author (
+        id INT GENERATED ALWAYS AS IDENTITY,
+        first_name VARCHAR(30) NOT NULL,
+        last_name VARCHAR(30) NOT NULL,
+        PRIMARY KEY(id)
+    );
+
+-- This table stores information about the relationship between games and authors
+
+CREATE TABLE
+    Creation(
+        item_id INT,
+        author_id INT,
+        PRIMARY KEY(item_id, author) -- A constraint to ensures that every author/item exists in the related table
+        CONSTRAINT FK_author FOREIGN KEY (author_id) REFERENCES Author (id) CONSTRAINT FK_item FOREIGN KEY (item_id) REFERENCES Game (id)
+    );
+
+CREATE TABLE
+    genres (
+        id INT GENERATED ALWAYS AS IDENTITY,
+        name VARCHAR(255) NOT NULL,
+        PRIMARY KEY(id)
+    );
+
+CREATE TABLE
+    music_albums (
+        id INT GENERATED ALWAYS AS IDENTITY,
+        title VARCHAR(255) NOT NULL,
+        publish_date DATE NOT NULL,
+        archived BOOLEAN NOT NULL,
+        on_spotify BOOLEAN NOT NULL,
+        genre_id INT DEFAULT NULL,
+        -- Foreign key that references genres table
+        source_id INT DEFAULT NULL,
+        -- Assuming that there will be a sources table
+        label_id INT DEFAULT NULL,
+        -- Assuming that there will be a labels table
+        author_id INT DEFAULT NULL,
+        -- Foreign key that references Author table
+        PRIMARY KEY(id),
+        CONSTRAINT FK_genre FOREIGN KEY (genre_id) REFERENCES genres (id),
+        CONSTRAINT FK_source FOREIGN KEY (source_id) REFERENCES sources (id),
+        CONSTRAINT FK_label FOREIGN KEY (label_id) REFERENCES labels (id),
+        CONSTRAINT FK_author FOREIGN KEY (author_id) REFERENCES Author (id)
+    );
+
+CREATE TABLE Creation(
+    item_id INT,
+    author_id INT,
+    PRIMARY KEY(item_id, author) -- A constraint to ensures that every author/item exists in the related table
+    CONSTRAINT FK_author FOREIGN KEY (author_id) REFERENCES Author (id)
+    CONSTRAINT FK_item FOREIGN KEY (item_id) REFERENCES Game (id)
+);
+
+CREATE TABLE Book (
+    id INT GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY NOT NULL,
+    publisher VARCHAR(50),
+    cover_state VARCHAR(50),
+    archieved BOOLEAN NOT NULL,
+    publish_date DATE NOT NULL,
+    item_id INT NOT NULL
+);
+    
+CREATE TABLE Label (
+    id INT GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY NOT NULL,
+    title VARCHAR(255),
+    color VARCHAR(100),
+    publish_date DATE NOT NULL,
+    item_id INT NOT NULL
+    book_id INT NOT NULL
+)
+
+-- schema for the 'movies' table
+CREATE TABLE movies (
+    id SERIAL PRIMARY KEY,
+    label VARCHAR(255) NOT NULL,
+    author VARCHAR(255),
+    genre VARCHAR(255) NOT NULL,
+    publish_date DATE NOT NULL,
+    silent BOOLEAN NOT NULL,
+    archived BOOLEAN NOT NULL,
+    source_id INT,
+    FOREIGN KEY (source_id) REFERENCES sources (id)
+);
```

Comment:
- [ ] Please add foreign key for item_id in movies table and you don't need source_id foreign key here as it does not have direct relation with source table